Source/Core/Core/HW/DSPHLE/UCodes/AX.h

```diff
@@ -141,6 +142,33 @@ class AXUCode : public UCodeInterface
   virtual void HandleCommandList();
   void SignalWorkEnd();
 
+  struct BufferDesc
+  {
+    int* ptr;
+    int samples_per_milli;
+  };
+
+  template <int Millis, size_t BufCount>
+  void InitMixingBuffers(u32 init_addr, const std::array<BufferDesc, BufCount>& buffers)
+  {
+    std::array<u16, 3 * BufCount> init_array;
+    Memory::CopyFromEmuSwapped(init_array.data(), init_addr, sizeof(init_array));
+    for (int i = 0; i < (int)BufCount; ++i)
+    {
+      const BufferDesc& buf = buffers[i];
+      s32 value = (u32(init_array[3 * i]) << 16) | init_array[3 * i + 1];
```

Comment:
```suggestion
      auto value = s32((u32(init_array[3 * i]) << 16) | init_array[3 * i + 1]);
```

Otherwise this is implicitly sign converting